File: solutions/python/matrix/2/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.rows = [[int(n) for n in row.split()] for row in matrix_string.splitlines()]

    def row(self, index):
        return self.rows[index - 1]

    def column(self, index):
        return [row[index - 1] for row in self.rows]

File: solutions/ruby/matrix/1/matrix.rb
class Matrix
  def initialize(matrix_str)
    @rows = matrix_str.split("\n").map { |row| row.split.map(&:to_i) }
    @columns = @rows.transpose
  end

  def row(i)
    @rows[i - 1]
  end

  def column(i)
    @columns[i - 1]
  end
end
